internal/pkg/agent/configuration/fleet_server.go

```diff
@@ -19,7 +19,7 @@ type FleetServerConfig struct {
 	Host         string                   `config:"host" yaml:"host,omitempty"`
 	Port         uint16                   `config:"port" yaml:"port,omitempty"`
 	InternalPort uint16                   `config:"internal_port" yaml:"internal_port,omitempty"`
-	TLS          *tlscommon.Config        `config:"ssl" yaml:"ssl,omitempty"`
+	TLS          *tlscommon.ServerConfig  `config:"ssl" yaml:"ssl,omitempty"`
```

Comment:
I don't know if there's anything else needed as far as bootstrapping goes when it comes to passing tls config to the fleet-server instance

Reply:
I think this has an issue because the [ServerConfig struct](https://github.com/elastic/elastic-agent-libs/blob/main/transport/tlscommon/server_config.go#L29) lacks the yaml tags that the Config struct has